torch/distributed/_device_mesh.py

```diff
@@ -230,11 +230,10 @@ def _validate_mesh(self):
 
         # validate that all calling ranks pass in the same `mesh` argument.
         self_mesh = self.mesh.to(self.device_type).contiguous()
-        mesh_tensor = funcol.all_gather_tensor(
-            self_mesh, gather_dim=0, group=_get_default_group()
-        )
-        mesh_tensor_chunked = torch.chunk(mesh_tensor, get_world_size())
-        for other_rank, other_mesh in enumerate(mesh_tensor_chunked):
+        tensor_list = [torch.zeros_like(self_mesh) for _ in range(get_world_size())]
+        all_gather(tensor_list, self_mesh)
```

Comment:
@wz337 I remembered @voznesenskym changed this to functional collective for some torch compile FSDP enablement, I am not sure if we change this to non-functional one would cause some trouble.